drafts/2020-02-20-Rest-client-generator.adoc

```diff
@@ -0,0 +1,133 @@
+don't forget any of the details.
+
+== Example
+Say you own a Java web application that you implemented with Microprofile and
+as a marketplace it serves as the front end for a collection of stores.
+Today you are going to add a pet store to the marketplace. You found a pet store back-end service
+that you're going to use because it provides a RESTful interface.
+The API allows people to list their pets for sale if they provide its name and links
+to some photos of the pet. In response, the API creates an id number for the listing.
+You can use the OpenAPI Generator to accelerate the integration of this service into the marketplace.
+
+The pet store service documents the API using the OpenAPI specification because it is both human readable and machine readable and the developer provided the API documentation in a file called petstore.yaml. The challenge now is to design and code the Java module that will communicate with the pet store service.
+
+The OpenAPI Generator can write some of that code for you. It can interpret the description in the yaml file and pull out the data model and the API. It will create Java files that describe the data model objects and method interfaces for each of the pet store endpoints.
+
+== The OpenAPI Description
+As an example the following snippet shows an endpoint to add a new pet in the petstore.yaml:
+-----
+/pet:
+  post:
+    summary: Add a new pet to the store
+    consumes:
+      - application/json
+    produces:
+      - application/json
+    parameters:
+      - in: body
+        name: body
+        schema:
+          $ref: '#/definitions/Pet'
+-----
+This endpoint's URL could look something like https://<server>/pet. The http verb is POST, the request's body is JSON and the body conforms to the schema referenced in the 'definitions' portion of the API document:
+-----
+  Pet:
+    description: A pet for sale in the pet store
+    type: object
+    properties:
+      id:
+        type: integer
+      name:
+        type: string
+        example: Rex the dog
+      photoUrls:
+        type: array
+        items:
+          type: string
+-----
+This schema defines three fields: one integer, one string and one array. A sample Pet object (in yaml format) could be:
+-----
+Pet:
+  id: 1234
+  name: Rex
+  photoUrls:
+    - https://myphotos.com/abcd1234
+-----
+== See Generated Code
+The generator will create model objects in Java for each of the types defined in the petstore.yaml. You can use these objects to transfer data to and from the service. For example, the Java Pet object will be, in part:
+-----
+public class Pet  {
+  private Long id;
+  private String name;
+  private List<String> photoUrls = new ArrayList<String>();
+  ...
+-----
+The generator will also create a Java interface for each endpoint. You can use these methods in your Microprofile front end to access the service. For example, to create a new pet you would call this method:
```

Comment:
`Microprofile` -> `MicroProfile`
Is "front end" the appropriate term here (since "front end" might imply UI rather than server code)?  Should it just be "your MicroProfile application"?